bitbots_vision:
  ros__parameters:
    # Defining the default vision parameters
    component_ball_detection_active: true # activate the ball detection
    component_camera_cap_check_active: true # activate a one-time check whether the camera cap is still on the camera
    component_debug_image_active: false # activate the debug image
    component_field_detection_active: true # activate the field detection
    component_goalpost_detection_active: false      # activate the goalpost detection
    component_line_detection_active: true           # activate the line detection
    component_obstacle_detection_active: true       # activate the obstacle detection

    vision_blind_threshold: 30 # Brightness threshold under which the vision thinks, that someone forgot the camera cap

    ROS_audio_msg_topic: "speak" # ROS topic of the audio message
    ROS_img_msg_topic: "camera/image_proc" # ROS topic of the image message
    ROS_ball_msg_topic: "balls_in_image" # ROS topic of the ball message
    ROS_goal_posts_msg_topic: "goal_posts_in_image" # ROS topic of the goal posts message
    ROS_obstacle_msg_topic: "robots_in_image" # ROS topic of the obstacles message
    ROS_line_msg_topic: "line_in_image" # ROS topic of the line message
    ROS_line_mask_msg_topic: "line_mask_in_image" # ROS topic of the line mask message
    ROS_debug_image_msg_topic: "debug_image" # ROS topic of the debug image message
    ROS_field_mask_image_msg_topic: "field_mask" # ROS topic of the field mask debug image message

    yoeo_model_path: '2022_10_07_flo_torso21_yoeox'
    yoeo_nms_threshold: 0.4  # Non-maximum suppression threshold
    yoeo_conf_threshold: 0.5  # YOEO confidence threshold
    yoeo_framework: 'tvm'  # The neural network framework that should be used ['openvino', 'onnx', 'pytorch', 'tvm']

    ball_candidate_rating_threshold: 0.5 # A threshold for the minimum candidate rating
    ball_candidate_max_count: 1 # The maximum number of balls that should be published

    red_color_detector_h_zero_crossing: true # Whether the red color detector interval includes the hue zero-crossing
    red_color_detector_lower_values_h: 146 # Lower bound for the red color detector hue
    red_color_detector_lower_values_s: 160 # Lower bound for the red color detector saturation
    red_color_detector_lower_values_v: 20 # Lower bound for the red color detector value/brightness
    red_color_detector_upper_values_h: 10 # Upper bound for the red color detector hue
    red_color_detector_upper_values_s: 215 # Upper bound for the red color detector saturation
    red_color_detector_upper_values_v: 220 # Upper bound for the red color detector value/brightness

    blue_color_detector_h_zero_crossing: false # The blue color interval does not contain the hue zero-crossing
    blue_color_detector_lower_values_h: 90 # Lower bound for the blue color detector hue
    blue_color_detector_lower_values_s: 140 # Lower bound for the blue color detector saturation
    blue_color_detector_lower_values_v: 60 # Lower bound for the blue color detector value/brightness
    blue_color_detector_upper_values_h: 100 # Upper bound for the blue color detector hue
    blue_color_detector_upper_values_s: 170 # Upper bound for the blue color detector saturation
    blue_color_detector_upper_values_v: 225 # Upper bound for the blue color detector value/brightness

    obstacle_finder_method: "convex" # Method for the obstacle finder (distance, convex or step)
    obstacle_color_threshold: 4 # An obstacle is defined as blue/red if it contains more blue or red than this threshold (amount of pixels in per mille)
    obstacle_candidate_min_width: 15 # Minimum width of an obstacle
    obstacle_candidate_max_width: 150 # Maximum width of an obstacle
    obstacle_finder_step_length: 5 # Length of an object detection step along the field boundary
    obstacle_finder_value_increase: 2.0 # Factor of the impact of the height of the field boundary on the distance threshold

    caching: true # Used to deactivate caching for profiling reasons
